platforms/ide/problems-api/src/main/java/org/gradle/api/problems/internal/DefaultReportableProblem.java

```diff
@@ -65,4 +66,25 @@ public void setProblemService(InternalProblems problemService) {
     public void report() {
         problemService.reportAsProgressEvent(this);
     }
+
+    @Override
+    public UnboundReportableProblemBuilder toBuilder() {
+        return new DefaultReportableProblemBuilder(problemService, this);
+    }
+
+    @Override
+    public boolean equals(@Nullable Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        return this.problemService == ((DefaultReportableProblem) o).problemService && super.equals(o);
+    }
+
+    @Override
+    public int hashCode() {
+        return problemService.hashCode() + super.hashCode();
```

Comment:
🤔 It's surprising to see a non-standard `hashCode` implementation. Would it have been possible to use something easily recognizable like `Objects.hashCode(...)` ?

Reply:
I overlooked those :(

I'll address it next time we touch it, which will be pretty soon :)
Thx for the feedback